hbase-server/src/main/resources/hbase-webapps/regionserver/rsNamedQueueLog.jsp

```diff
@@ -0,0 +1,157 @@
+<%--
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+--%>
+<%@ page contentType="text/html;charset=UTF-8"
+  import="java.util.Date"
+  import="java.util.List"
+  import="java.util.Arrays"
+  import="java.util.HashSet"
+  import="org.apache.hadoop.conf.Configuration"
+  import="org.apache.hadoop.hbase.client.Admin"
+  import="org.apache.hadoop.hbase.client.SnapshotDescription"
+  import="org.apache.hadoop.hbase.http.InfoServer"
+  import="org.apache.hadoop.hbase.master.HMaster"
+  import="org.apache.hadoop.hbase.snapshot.SnapshotInfo"
+  import="org.apache.hadoop.util.StringUtils"
+  import="org.apache.hadoop.hbase.TableName"
+  import="org.apache.hadoop.hbase.client.ServerType"
+  import="org.apache.hadoop.hbase.client.LogEntry"
+  import="org.apache.hadoop.hbase.regionserver.HRegionServer"
+  import="org.apache.hadoop.hbase.client.OnlineLogRecord"
+  import="org.apache.hadoop.hbase.HConstants"
+  import="org.apache.hadoop.hbase.ServerName"
+%>
+<%
+  HRegionServer rs = (HRegionServer) getServletContext().getAttribute(HRegionServer.REGIONSERVER);
+  Configuration conf = rs.getConfiguration();
+  List<OnlineLogRecord> slowLogs = null;
+  List<OnlineLogRecord> largeLogs = null;
+
+  if(rs.isOnline()) {
+    try (Admin rsAdmin = rs.getConnection().getAdmin()) {
```

Comment:
Is it possible to also use the NmaedQueueRecorder directly on branch-2?

Reply:
A few days ago, I wanted to wait for the master branch to be approved before pushing branch-2. Deal with it right now.